### PR TITLE
Fix regression environment

### PR DESCRIPTION
**Pull request by stefanpenner**
Original PR: https://github.com/ember-engines/ember-engines/pull/538 I accidentally pushed master to that branches master, causing GH to auto-close the branch.

Anyways, I have restored the commits here and am working to close out the remaining issues now, so we can land this.

- [x] manual test
- [x] more automated tests

### CODE REVIEW

**Comment by villander** [comment]
Thanks for the help @stefanpenner

File: node-tests/acceptance/build-test.js [hunk 1]
```diff
@@ -73,10 +73,7 @@ describe('Acceptance', function() {
         expect(output.manifest()).to.deep.equal(expectedManifests['lazy']);
         output.contains('assets/node-asset-manifest.js');
         // Engine inherites its environment from the parent
-        output.contains(
-          `engines-dist/${engineName}/config/environment.js`,
-          /"environment":"development"/
-        );
+        output.contains(`engines-dist/${engineName}/config/environment.js`);
```
Comment (stefanpenner) [inline node-tests/acceptance/build-test.js]: I'll need to expand the tests, to support  multiple scenarios.

File: lib/engine-addon.js [hunk 2]
```diff
@@ -910,17 +903,24 @@ module.exports = {
 
       configPath = path.join(this.root, configPath, 'environment.js');
 
+      let key = `${configPath}|${env}`;
```
Comment (stefanpenner) [inline lib/engine-addon.js]: @rwjblue can i get your 👀  on this?

**Comment by stefanpenner** [comment]
@villander I got a wall of meetings coming up, but I will drive this to completion this evening.

File: node-tests/acceptance/build-test.js (hunk 1 above)
Comment (villander) [reply node-tests/acceptance/build-test.js]: cc: @stefanpenner

**Comment by villander** [comment]
@stefanpenner you has any updates about this PR?

**Comment by villander** [comment]
@stefanpenner we can merge this?

**Comment by stefanpenner** [comment]
rebased, now that tests all run.

**Comment by lennyburdette** [comment]
I'm running into issues with the incorrect engine environment. I've been using a branch with this change cherry-picked and it's working fine. Can it be merged as is? Or would it be helpful if I tried to write some more tests?

**Comment by villander** [comment]
@stefanpenner when we can merge this? Unfortunately I do not know how to write tests for this feature.

**Comment by stefanpenner** [comment]
Sorry for the delay on this one. I had some other blocking work that prevented me from testing this in the primary app I need to test this on.

**Comment by stefanpenner** [comment]
I apparently don't have publish permissions, let me track those down.

**Comment by stefanpenner** [comment]
released as v0.5.21 🎉

**Pull request merged**